src/core/nodeOps.ts

```diff
@@ -39,7 +39,8 @@ export const nodeOps: RendererOptions<TresObject, TresObject> = {
       if (props?.object === undefined) logError('Tres primitives need a prop \'object\'')
       const object = props.object as TresObject
       name = object.type
-      instance = Object.assign(object, { type: name, attach: props.attach, primitive: true })
+      instance = Object.assign(object.clone(), { type: name }) as TresObject
```

Comment:
I'm not sure if there's a workaround, but using a clone here means that references in the component's setup won't work.

I've added a comment to playground primitives.vue file with some code that demonstrates the problem.

Reply:
@andretchen0 Thats a good point thanks for raising it, the problem with the reference comes when we try to make the `:object` reactive to swap the instance, which completely changes the ref.

Not sure how to deal with that.